Flaky test: TestPollConfig_ReportsOnlyTheSettledContent fails when its own writer stalls

`TestPollConfig_ReportsOnlyTheSettledContent` failed the `test (ubuntu-latest)` job on `main` for the v2.15.0 release commit (`1fcbf5b`), run [33641652407](https://github.com/nuetzliches/hookaido/actions/runs/33641652407):

```
--- FAIL: TestPollConfig_ReportsOnlyTheSettledContent (0.41s)
    watch_poll_test.go:413: reload observed a partial config: "# v2\n"
```

Every other job in that run was green, as was the identical PR run on `chore/release-v2.15.0`, and the test passes locally on repeat runs. `pollConfig` did nothing wrong here — the test's own timing assumption is what broke.

### The test blames the poller for a stall in its own writer

`internal/app/watch_poll_test.go:354` writes the new config in three chunks separated by `pollSettleDelay / 5` (40 ms), so that a poll lands inside the write while the file is still changing across a settle window. The assertion is absolute: any content other than the final one that reaches the reload callback fails the test.

That only holds while the writer keeps up. If the runner stretches one of those sleeps past `pollSettleDelay` (200 ms, `internal/app/watch_poll.go:24`) — plausible on a loaded `-race` runner — then `"# v2\n"` genuinely held still across the settle window. `readSettledConfigHash` is *correct* to report it at that point, and the full config is still reported afterwards once it settles too. The test fails anyway, because it cannot tell a real premature report from its own scheduling gap.

This is the second occurrence of the same failure mode. 738a67f (#292) already widened the margin, from a flat 120 ms to `pollSettleDelay / 5`, and left a comment at `internal/app/watch_poll_test.go:379` explaining exactly this hazard. A wider margin lowers the probability; it does not remove the class, and the release commit hit it again.

### Fix direction

Judge a partial report against the writer's actual timeline instead of trusting the sleeps. Record when each chunk hit the disk, and treat a reported prefix as a failure only when the *next* chunk followed it within `pollSettleDelay` — i.e. when the file demonstrably changed again inside the settle window, so differing reads were guaranteed and the report was premature. If the gap after that prefix exceeded the settle window, the writer stalled, the prefix legitimately settled, and the test should log that and keep waiting for the final content rather than fail.

The timeline is recorded by the writing goroutine and evaluated only after it finishes (reports arrive through a buffered channel), so this needs no extra synchronisation. The assertion that matters — a prefix reported while the file was still being written is a bug — stays intact, and the test still requires the settled config to be reported.

--- a/internal/app/watch_poll_test.go
+++ b/internal/app/watch_poll_test.go
@@ -376,17 +376,20 @@ func TestPollConfig_ReportsOnlyTheSettledContent(t *testing.T) {
 	// (the gap exceeds the poll interval) but fast enough that the file is still
 	// changing across a settle window.
 	//
-	// The gap is a fraction of pollSettleDelay rather than a bare duration,
-	// because the relationship is the whole assumption of the test: at a flat
-	// 120ms against a 200ms settle the margin was thin enough that a loaded
-	// `-race` runner could stretch one sleep past the window, at which point the
-	// poller correctly reported settled partial content and the test blamed it
-	// for a stall in the writer.
+	// The gap is a fraction of pollSettleDelay rather than a bare duration, but
+	// the gap alone cannot carry the test: a loaded `-race` runner can stretch
+	// one of these sleeps past the settle window whatever it is set to, and then
+	// a prefix really did hold still. So record when each chunk hit the disk and
+	// judge a partial report against that timeline below, instead of assuming
+	// the sleeps were timely.
 	const chunkGap = pollSettleDelay / 5
 	f, err := os.Create(path)
 	if err != nil {
 		t.Fatalf("create: %v", err)
 	}
+	var prefixes []string
+	var writtenAt []time.Time
+	content := ""
 	for _, chunk := range []string{"# v2\n", "route \"/a\" {}\n", "route \"/b\" {}\n"} {
 		if _, err := f.WriteString(chunk); err != nil {
 			t.Fatalf("write chunk: %v", err)
@@ -394,13 +397,19 @@ func TestPollConfig_ReportsOnlyTheSettledContent(t *testing.T) {
 		if err := f.Sync(); err != nil {
 			t.Fatalf("sync: %v", err)
 		}
+		content += chunk
+		prefixes = append(prefixes, content)
+		writtenAt = append(writtenAt, time.Now())
 		time.Sleep(chunkGap)
 	}
 	if err := f.Close(); err != nil {
 		t.Fatalf("close: %v", err)
 	}
 
-	want := "# v2\nroute \"/a\" {}\nroute \"/b\" {}\n"
+	// The reports arrive through a buffered channel and are read only now that
+	// the writing goroutine is done with the timeline, so no synchronisation is
+	// needed around it.
+	want := prefixes[len(prefixes)-1]
 	deadline := time.After(5 * time.Second)
 	for {
 		select {
@@ -408,11 +417,31 @@ func TestPollConfig_ReportsOnlyTheSettledContent(t *testing.T) {
 			if got == want {
 				return
 			}
-			// Anything else means a partial state was reported as a config
-			// change, which is exactly what the settle exists to prevent.
+			// A partial state reported as a config change is exactly what the
+			// settle exists to prevent -- but only when the file demonstrably
+			// changed again inside the settle window, which is what makes the
+			// poller's two reads differ and the report premature. If the writer
+			// stalled longer than that after this prefix, the prefix genuinely
+			// settled: reporting it is correct, and the full config is still
+			// reported once it settles too.
+			if i := indexOfString(prefixes, got); i >= 0 && i+1 < len(writtenAt) {
+				if stall := writtenAt[i+1].Sub(writtenAt[i]); stall >= pollSettleDelay {
+					t.Logf("writer stalled %v after chunk %d, so %q had settled; still waiting for the full config", stall, i+1, got)
+					continue
+				}
+			}
 			t.Fatalf("reload observed a partial config: %q", got)
 		case <-deadline:
 			t.Fatal("the settled config was never reported")
 		}
 	}
+}
+
+func indexOfString(values []string, want string) int {
+	for i, v := range values {
+		if v == want {
+			return i
+		}
+	}
+	return -1
 }

